ipn/ipnserver/server.go

```diff
@@ -238,12 +238,26 @@ func bufferHasHTTPRequest(br *bufio.Reader) bool {
 		mem.Contains(mem.B(peek), mem.S(" HTTP/"))
 }
 
+// bufferIsConnect reports whether br looks like it's likely an HTTP
+// CONNECT request.
+func bufferIsConnect(br *bufio.Reader) bool {
+	peek, _ := br.Peek(br.Buffered())
+	return mem.HasPrefix(mem.B(peek), mem.S("CONN"))
```

Comment:
I wish the compiler was good enough to make `strings`/`bytes` as nice as `mem` to use here.

Reply:
Indeed.